tutorials/rendering/compositor.rst

```diff
@@ -0,0 +1,368 @@
+                var z_groups = 1
+
+                # Push constant
+                var push_constant : PackedFloat32Array = PackedFloat32Array()
+                push_constant.push_back(size.x)
+                push_constant.push_back(size.y)
+                push_constant.push_back(0.0)
+                push_constant.push_back(0.0)
+
+                # Loop through views just in case we're doing stereo rendering. No extra cost if this is mono.
+                var view_count = render_scene_buffers.get_view_count()
+                for view in range(view_count):
+                    # Get the RID for our color image, we will be reading from and writing to it.
+                    var input_image = render_scene_buffers.get_color_layer(view)
+
+                    # Create a uniform set, this will be cached, the cache will be cleared if our viewports configuration is changed.
+                    var uniform : RDUniform = RDUniform.new()
+                    uniform.uniform_type = RenderingDevice.UNIFORM_TYPE_IMAGE
+                    uniform.binding = 0
+                    uniform.add_id(input_image)
+                    var uniform_set = UniformSetCacheRD.get_cache(shader, 0, [ uniform ])
+
+                    # Run our compute shader.
+                    var compute_list := rd.compute_list_begin()
+                    rd.compute_list_bind_compute_pipeline(compute_list, pipeline)
+                    rd.compute_list_bind_uniform_set(compute_list, uniform_set, 0)
+                    rd.compute_list_set_push_constant(compute_list, push_constant.to_byte_array(), push_constant.size() * 4)
+                    rd.compute_list_dispatch(compute_list, x_groups, y_groups, z_groups)
+                    rd.compute_list_end()
+
+At the start of this method we check if we have a rendering device,
+if our callback type is the correct one, and check if we have our shader.
+
+.. note::
+
+    The check for the effect type is only a safety mechanism.
+    We've set this in our ``_init`` function, however it is possible
+    for the user to change this in the UI.
+
+Our ``p_render_data`` parameter gives us access to an object that holds
+data specific to the frame we're currently rendering. We're currently only
+interested in our render scene buffers, which provide us access to all the
+internal buffers used by the rendering engine.
+Note that we cast this to :ref:`RenderSceneBuffersRD <class_RenderSceneBuffersRD>`
+to expose the full API to this data.
+
+Next we obtain our ``internal size`` which is the resolution of our 3D render
+buffers before they are upscaled (if applicable), upscaling happens after our
+post processes have run.
+
+From our internal size we calculate our group size, see our local size in our
+template shader.
+
+We also populate our push constant so our shader knows our size.
+Godot does not support structs here **yet** so we use a
+``PackedFloat32Array`` to store this data into. Note that we have
+to pad this array correctly.
```

Comment:
```suggestion
We also populate our push constant so our shader knows our size.
Godot does not support structs here **yet** so we use a
``PackedFloat32Array`` to store this data into. Note that we have
to pad this array with a 16 byte alignment. In other words, the
length of our array needs to be a multiple of 4.
```

Reply:
Owh thanks, that is definitely clearer. I know that for uniform buffers we have even stricter alignment rules but I wasn't sure how much applies to a push constant.